File: server/dotnet/sdk-dotnet-example/Business/Domain/Payments/PaymentDetails/OperationOutput.cs
namespace Business.Domain.Payments.PaymentDetails;

public class OperationOutput
{
    public AmountOfMoney? AmountOfMoney { get; set; }

    public string? Id { get; set; }

    public OperationPaymentReferences? OperationReferences { get; set; }

    public string? PaymentMethod { get; set; }

    public PaymentReferences? References { get; set; }

    public string? Status { get; set; }

    public PaymentStatusOutput? StatusOutput { get; set; }
}

File: server/php/src/Application/Domain/Payments/PaymentDetails/OperationOutput.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class OperationOutput
{
    public ?AmountOfMoney $amountOfMoney = null;
    public ?string $id = null;
    public ?OperationPaymentReferences $operationReferences = null;
    public ?string $paymentMethod = null;
    public ?PaymentReferences $references = null;
    public ?string $status = null;
    public ?PaymentStatusOutput $statusOutput = null;

    public function toArray(): array
    {
        return [
            'amountOfMoney' => $this->amountOfMoney?->toArray(),
            'id' => $this->id,
            'operationReferences' => $this->operationReferences?->toArray(),
            'paymentMethod' => $this->paymentMethod,
            'references' => $this->references?->toArray(),
            'status' => $this->status,
            'statusOutput' => $this->statusOutput?->toArray(),
        ];
    }
}
